lib/web.ts

```diff
@@ -158,6 +164,7 @@ export default class AutoConsent {
     parseDeclarativeRules(declarativeRules: RuleBundle) {
         if (declarativeRules.consentomatic) {
             Object.keys(declarativeRules.consentomatic).forEach((name) => {
+                // @ts-expect-error - consentomatic is defined at this point
```

Comment:
TypeScript can’t do type narrowing across function boundaries. We can help it out by using `for … in` instead.

```ts
 for (const name in declarativeRules.consentomatic) {
    this.addConsentomaticCMP(name, declarativeRules.consentomatic[name]);
}
```